Move files to proper locations. Add idea iml file. Add gradlew

dankwart-de opened the pull request.

Commit:
Move files to proper locations. Add idea iml file. Add gradlew

fkaercher reviewed (commented):
Why did you add idea iml and graddle wrapper etc?

dankwart-de commented:
When you want to use gradlew, it is recommended to add the gradlew (and gradle?) files to the repository. That's what I found on their website.

I added the RoddyToolLib project to our RoddyInfra project map. For this, the IML is necessary. That's also much closer to the structure we already have.

I also moved the files from src/main/groovy to src and the test files to a test folder. Idea uses this setup and struggled with the former one.

dankwart-de commented:
You can do the same things for BatchEuphoria! Be aware, that I also renamed the IOHelper class back to RoddyIOHelper. I want to keep things with as few changes as possible now. I know, the suggestion came from me, but I know think it is much better to create as few trouble as possible :)

Commit:
Moved files out of subpackages.

Sorry for this mess. But I thought really hard about it and decided to
recreate the original package structure. The problem is, that with all
the new packages, a huge API break will occur. Not all of the classes
will create problems, but things like the IOHelper stuff is used in
workflows. The structure proposed by Florian was good and logical and I
think we can do this with the next Roddy release. I don't think, we will
have too many customers for the library in the near future. In regards
to the cluster lib, I think it is safe for this. The cluster lib has
some dependencies to the Tool Lib but that will not create problems.

Commit:
Recreated gradle file structure

vinjana commented on src/main/groovy/de/dkfz/roddy/StringConstants.java:
```diff
@@ -33,7 +33,7 @@
     public static final String SPLIT_SBRACKET_RIGHT = "\\[";
 
-    public static final String STOP = ".";
+    public static final String STOP = "";
```
What is the purpose of an empty string?

Commit:
Pre Weekend commit

Commit:
Change Executionhelper to use "bash -c" to execute shell commands

This was formerly in BashCommandSet but now, as there is RoddyToolLib,
we need to find a good way to handle this. For now, we will just stick
to hardcoded "bash -c". Otherwise most of the shell commands won't work.

Commit:
Change Executionhelper to use "bash -c" to execute shell commands

This was formerly in BashCommandSet but now, as there is RoddyToolLib,
we need to find a good way to handle this. For now, we will just stick
to hardcoded "bash -c". Otherwise most of the shell commands won't work.

vinjana committed:
Groovy 2.4.7 change back (for Gradle in Roddy). Travis file.

vinjana committed:
Added sudo: false to .travis.yml.

vinjana committed:
Fixed Gradle and Java versions in the build.gradle.

vinjana committed:
Fixed JDK version for testing.

vinjana committed:
Hmm, better read the docs. Travis does not support OpenJDK 8.

dankwart-de committed:
Merge pull request #4 from eilslabs/travis

Travis

vinjana committed:
Added group de.dkfz.eilslabs.

Commit:
Change STOP="" to STOP="." in StringConstants

Commit:
Merge branch 'develop' of https://github.com/eilslabs/RoddyToolLib into develop

dankwart-de commented:
Could you please re-review?

vinjana commented on src/main/groovy/de/dkfz/roddy/core/InfoObject.java:
```diff
@@ -67,6 +68,7 @@ public void updateTimeStamp() {
 
     public int getAgeInSeconds() {
         long age = (int)(((new Date().getTime()) - timeStamp.getTime()) / 1000);
+//        System.out.println(age);
```
Debug code.

dankwart-de replied:
Done

vinjana commented on src/main/groovy/de/dkfz/roddy/execution/io/ExecutionResult.java:
```diff
@@ -0,0 +1,51 @@
+     * First line of the result array.
+     * Null if no entries are in the array.
+     */
+    public final String firstLine;
```
Why is this necessary? Why is this not a method? Also, testing no output by `resultLines.empty()` is much more speaking than `firstLine == null`.

dankwart-de replied:
One of my many examples for optimizations :). If you have an accessor and repeatedly call it, I tend to cache values. Same here... though we don't need the cache.

vinjana replied:
In this case, I doubt the optimization will actually yield an improvement -- neither for the emptyness check nor for getting the first line is an expensive operation (O(1)).

dankwart-de replied:
I'll just change it.

vinjana commented on src/main/groovy/de/dkfz/roddy/tools/RoddyIOHelperMethods.groovy:
```diff
@@ -162,16 +163,17 @@ class RoddyIOHelperMethods {
+            tgt.setExecutable(src.canExecute())
+
         } catch (Exception ex) {
             logger.severe(ex.toString())
```
Is this actually the right place to decide, whether a copy failure should be fatal or not? Is that not rather the responsibility of the calling code? It depends on the directory, whether a failure here is a problem.

dankwart-de replied:
Well, I know it is different because of the original non-strict Roddy design: "Run as far as you can". In that case it makes sense, but yes for a more public design we will take out the exception here.

vinjana replied:
Maybe we can add a switch to turn off (not on!) the exception. Thus the function can be used in both contexts -- strict (with exception) and non-strict.
